packages/server/worker/src/lib/job-polling.ts

```diff
@@ -0,0 +1,22 @@
+import {  ApQueueJob, QueueName,  system, WorkerSystemProps } from '@activepieces/server-shared'
+import { Semaphore } from 'async-mutex'
+import { workerApiService } from './api/server-api.service'
+
+const POLLING_CONCURRENCY = system.getNumberOrThrow(WorkerSystemProps.POLLING_CONCURRENCY)
+
+const pollLocks = {
+    [QueueName.ONE_TIME]: new Semaphore(POLLING_CONCURRENCY),
+    [QueueName.SCHEDULED]: new Semaphore(POLLING_CONCURRENCY),
+    [QueueName.WEBHOOK]: new Semaphore(POLLING_CONCURRENCY),
+}
+
+export async function pollJob(workerToken: string, queueName: QueueName): Promise<ApQueueJob | null> {
```

Comment:
It's better not to pollute the global namespace so the IDE doesn't suggest too many functions.

Put it inside a constant.

An example:
```javascript
const jobPolling = {
    // function definition here
    pollJobWithConnectionPool: function() {
        // function logic here
    }
};
export { jobPolling };
```